compiler/ballerina-lang/src/main/java/org/ballerinalang/util/diagnostic/DiagnosticErrorCode.java

```diff
@@ -807,7 +807,8 @@ public enum DiagnosticErrorCode implements DiagnosticCode {
 
     QUERY_CONSTRUCT_TYPES_CANNOT_BE_USED_WITH_COLLECT("BCS4049", "query.construct.types.cannot.be.used.with.collect"),
     VARIABLE_IS_SEQUENCED_MORE_THAN_ONCE("BCS4050", "variable.is.sequenced.more.than.once"),
-    INVALID_GROUPING_KEY_TYPE("BCS4051", "invalid.grouping.key.type")
+    INVALID_GROUPING_KEY_TYPE("BCS4051", "invalid.grouping.key.type"),
+    REST_ARG_CANNOT_BE_NAMED_ARG("BCE4052", "rest.arg.cannot.be.named.arg")
```

Comment:
Let's change the error message to `named argument not allowed for rest parameter` as it is more convicing to the user.

```suggestion
    NAMED_ARG_NOT_ALLOWED_FOR_REST_PARAM("BCE4052", "named.arg.not.allowed.for.rest.param")
```